Assets/Scripts/Army/Army.cs

```diff
@@ -2,19 +2,16 @@
 using System.Collections.Generic;
 using System.Collections.ObjectModel;
 using System.Linq;
-
 using Unity.Mathematics;
-
-
 using UnityEngine;
 using UnityEngine.Events;
 using Mirror;
-using System;
 
-public class Army : NetworkBehaviour
+[RequireComponent(typeof(ArmyMovement))]
+public class Army : Entity
 {
-    List<Unit> armyUnits = new List<Unit>(); // Change to set if necessary
-    public ReadOnlyCollection<Unit> ArmyUnits => armyUnits.AsReadOnly();
+    readonly SyncList<Unit> armyUnits = new SyncList<Unit>(); // Change to set if necessary
```

Comment:
This should work just as List<Unit>, but keep it as SyncList<Unit> to later sync client-side UI corresponding to army size